build(PKGBUILD): add Arch Linux package

norz3n opened the pull request:
Greetings, I decided to make a PKGBUILD to support Arch Linux builds for fun, I think you might be interested in it.

norz3n committed:
Packaging: Add Arch Linux support

BrycensRanch commented:
Hi, thank you for your PR. I've noticed some minor issues with it. For example, the packaging directory doesn't need additional folders for one file.  (the .spec file and PKGBUILD file should be right next to each other)

Additionally, the `metainfo` file is not only for RPM-based distros. It is used for software like `gnome-software` on Arch Linux and as per Arch Linux's packaging guidelines, it should be included for GNOME applications. <https://wiki.archlinux.org/title/GNOME_package_guidelines>

The Arch Linux package should use `./build.sh install --prefix /usr --dest-dir ${pkgdir} --skip compile` for installing files. This unifies packaging, as a fix to the Install Target a fix for all packages. 

Besides that, this PR looks great. Once the issues mentioned are ironed out, I will merge the pull request.

BrycensRanch commented on packaging/Arch/PKGBUILD:
```diff
@@ -0,0 +1,135 @@
+pkgdesc="Screenshot tool that handles images, text, and video (fork of ShareX)"
+arch=('x86_64' 'aarch64')
+url="https://github.com/BrycensRanch/SnapX"
+license=('GPL3')
```
The license should be `GPL-3.0-or-later` according to SPDX. Arch packaging guidelines expect this to be an SPDX license.
<https://wiki.archlinux.org/title/PKGBUILD#license>

norz3n committed:
packaging: Consolidate package build configurations for SnapX

- Add RPM spec file for Fedora/RHEL packaging
- Move Arch Linux PKGBUILD to root packaging directory
- Add desktop entry and AppStream metadata for Linux desktop integration
- Standardize package descriptions and dependencies across different package managers

BrycensRanch commented on packaging/PKGBUILD:
```diff
@@ -0,0 +1,103 @@
+        'curl'
+        'fontconfig'
+        'freetype2'
+        'openssl'
```
Upon building this PKGBUILD, it fails to build successfully on a fresh Arch Linux VM.

The build log states: 
https://pasteb.in/?bb64523c4bcaa2bc#752e5d2pkjJRXzJmfGhhzCg26DPTEPHVHVa5tTYFgw5


Additionally, the ./build.sh install target should only be invoked once as the `snapx` package is the base package where it wants to install files. This will prevent duplicate files. In the RPM spec file, the `snapx` package holds ownership of `/usr/share/SnapX` & `/usr/lib/snapx`, the Arch Linux package should mirror this.

BrycensRanch commented on packaging/PKGBUILD:
```diff
@@ -0,0 +1,103 @@
+    ./build.sh install --prefix /usr --dest-dir "${pkgdir}" --skip compile
+    
+    # Удаляем файлы, которые принадлежат другим пакетам
+    rm -f "${pkgdir}/usr/bin/snapx-gtk"
```
You don't need to do any install commands once you've invoked `./build.sh install` as it's done everything. All comments should be made in English to ensure clear communication for everyone.

BrycensRanch commented:
Also, I'd like to apologize for the delay. I've never used GitHub's "review" function yet. I reviewed your changes when they were made 6 hours ago then I realized why my review wasn't responded to. It wasn't marked as "finished"  😆

BrycensRanch commented:
Hello, @norz3n I have noticed there are problems with the install script such as it not creating a licenses directory and docs directory with the readme and PRIVACY.md. I am going to be addressing these problems tomorrow, I hope I can get a response regarding the package build failure with your current changes.

norz3n commented:
> Hello, @norz3n I have noticed there are problems with the install script such as it not creating a licenses directory and docs directory with the readme and PRIVACY.md. I am going to be addressing these problems tomorrow, I hope I can get a response regarding the package build failure with your current changes.

Okay. I'll fix that in the near future.

BrycensRanch commented:
Hi, @norz3n ! This pull request has been unblocked by commit 411e3ec70fd06c27fc59b7df5b59ab8ed6a1bee3 . Looking forward to an Arch Linux package. :)

BrycensRanch commented:
Thanks to commit https://github.com/BrycensRanch/SnapX/commit/7c4824424b24d3a40b98b925bf635a95432bfab7 SnapX's icons are now installed.

BrycensRanch commented:
Hi, the build log I provided here is currently going to expire in **11 days**. 

I've added the stale label as my reviews haven't been addressed. I want this pull request merged as recent crash reports from testers are considerably from Arch Linux users. 

I see you've permitted me to edit your PR. I am _open_ to all discussions, but if there is no activity within a week, I will submit a commit that will help ease my concerns and then **merge this PR and I continue as the maintainer of the AUR package** as hoped.
![image](https://github.com/user-attachments/assets/a7a9c288-2724-49e0-b593-6d1de8034efd)

BrycensRanch committed:
build(PKGBUILD): do not move binaries & only include metadata in snapx pkg

I also removed some of the makedepends. SnapX only needs .NET 9 & NativeAOT to build. 

Signed-off-by: Brycen G <brycengranville@outlook.com>

BrycensRanch committed:
build(PKGBUILD): replicate rpm packaging mistake

I know this is confusing, lol. However, all the SnapX binaries have dependency on the libraries in /usr/lib/snapx hence snapx-ui relies on the snapx package plus the NativeMessagingHost for the Chrome extension. I will push out an update to the packaging to fix these flaws.

Signed-off-by: Brycen G <brycengranville@outlook.com>

BrycensRanch commented:
I have made the necessary changes and the PKGBUILD now builds. Congrats.

![image](https://github.com/user-attachments/assets/a6b82aae-731d-4052-914f-cbba4a2fe46d)

BrycensRanch commented:
Hello, your work has been deployed to https://aur.archlinux.org/packages/snapx-ui ! Thank you for your contribution!

<img src="https://media3.giphy.com/media/v1.Y2lkPWJkM2VhNTdlNzhudzhkOTMzYnBkZjgyd2g1Nmh6ZHN1c3V1eXdyMm9hbmppeHVhMiZlcD12MV9naWZzX3NlYXJjaCZjdD1n/ghhynvHS4NbDG/giphy.gif"/>
